aten/src/ATen/native/cudnn/Conv_v7.cpp

```diff
@@ -821,6 +821,109 @@ void raw_cudnn_convolution_backward_weight_out(
   TORCH_INTERNAL_ASSERT(false, "This case should not be dispatched to cuDNN.");
 }
 
+void raw_cudnn_convolution_add_relu_out(
+    const Tensor& output,
+    const Tensor& input,
+    const Tensor& weight,
+    const Tensor& bias,
+    const Tensor& z,
+    float alpha,
+    IntArrayRef padding,
+    IntArrayRef stride,
+    IntArrayRef dilation,
+    int64_t groups) {
+  // This function is currently only called if frozen-model opt is turned on.
+  // The benchmarking and caching mechanism is not ready, so let's set benchmark
+  // to false.
+  bool benchmark = false;
+  // (!benchmark && !deterministic) means later try_all will call
+  // cudnnGetConvolutionForwardAlgorithm_v7 to pick the best convolution
+  // algorithm.
+  bool deterministic = false;
+  bool allow_tf32 = true;
+
+  auto dataType = getCudnnDataType(input);
+
+  ConvolutionArgs args{input, output, weight};
+  args.handle = getCudnnHandle();
+  setConvolutionParams(
+      &args.params,
+      input,
+      weight,
+      padding,
+      stride,
+      dilation,
+      groups,
+      deterministic,
+      allow_tf32);
+  args.idesc.set(input);
+  args.wdesc.set(
+      weight,
+      0,
+      input.suggest_memory_format() == at::MemoryFormat::ChannelsLast);
+  args.odesc.set(output);
+  args.cdesc.set(
+      dataType,
+      input.dim() - 2,
+      args.params.padding,
+      args.params.stride,
+      args.params.dilation,
+      args.params.groups,
+      args.params.allow_tf32);
+
+  TensorDescriptor zdesc;
+  zdesc.set(z);
+
+  // See Note [CuDNN broadcast padding].  Handle the left padding
+  // ourselves, but use TensorDescriptor's padding argument to do the rest.
+  TensorDescriptor bdesc;
+  bdesc.set(bias.expand({1, bias.size(0)}), output.dim());
+
+  ActivationDescriptor adesc;
+  adesc.set(CUDNN_ACTIVATION_RELU);
+
+  AlgoIterator<cudnnConvolutionFwdAlgoPerf_t>(args, benchmark)
```

Comment:
The equivalent pytorch eager code when we're not benchmarking is just:
```
    if (!benchmark) {
      AT_CUDNN_CHECK_WITH_SHAPES(cudnnGetConvolutionForwardAlgorithm_v7(
          args.handle,
          args.idesc.desc(),
          args.wdesc.desc(),
          args.cdesc.desc(),
          args.odesc.desc(),
          num_algos,
          &perf_count,
          perf_results.get()), args);
    } else { 
```

why are we doing the `try_all([&](const cudnnConvolutionFwdAlgoPerf_t` logic ?